### PR TITLE
OptionalJobProperty

### PR DESCRIPTION
**Pull request by jglick**
Simplification to allow `JobProperty` implementations to be spared the messy details of defining a UI for an optional property, which is the normal case.

Alternative to #1656.
- [X] new API
- [X] demonstrate simplification of `ParametersDefinitionProperty`

@reviewbybees

### CODE REVIEW

File: core/src/main/resources/hudson/model/Messages_es.properties [hunk 1]
```diff
@@ -311,3 +311,4 @@ UpdateCenter.DownloadButNotActivated=Descarga correcta. Se activar
 Computer.ConnectPermission.Description=Este permiso permite que los usuarios puedan conectar nodos o marcarlos como activos.
 BallColor.NotBuilt=Sin ejecutar.
 AbstractBuild_Building=Ejecutando.
+ParametersDefinitionProperty.DisplayName=Esta ejecuci�n debe parametrizarse
```
Comment (amuniz) [inline core/src/main/resources/hudson/model/Messages_es.properties]: `Esta ejecución es parametrizada` is a better translation.

**Comment by amuniz** [comment]
:bee: (as far as I understand the objective of this PR, since there is no JIRA ticket linked and only a short description: "new API", which is obvious).

**Comment by tfennelly** [comment]
:bee:

**Comment by jglick** [comment]
@reviewbybees done

**Comment by bpedersen2** [comment]
Seems like this change could be the cause of: https://issues.jenkins-ci.org/browse/JENKINS-31733

**Comment by bpedersen2** [comment]
https://issues.jenkins-ci.org/browse/JENKINS-31458

**Pull request merged**